src/Jenssegers/Mongodb/Eloquent/Builder.php

```diff
@@ -221,10 +222,18 @@ public function raw($expression = null)
         if ($results instanceof Cursor)
         {
             $results = iterator_to_array($results, false);
-
             return $this->model->hydrate($results);
         }
 
+        // Convert Mongo BSONDocument to a single object.
+        elseif ($results instanceof BSONDocument)
+        {
+            $results = $results->getArrayCopy();
+            $model = $this->model->newFromBuilder((array) $results);
+            $model->setConnection($this->model->getConnection());
+            return $model;
```

Comment:
You should be able to remove the setConnection call as newFromBuilder does this internally.